oneargopy/Argo.py

```diff
@@ -284,18 +278,58 @@ def trajectories(self, floats: int | list | dict)-> None:
         # Displaying graph
         plt.show()
 
-    def load_float_data(self, floats: int | list)-> None: 
+    def load_float_data(self, floats: int | list | dict, variables: str | list = None)-> pd: 
         """ A function to load float data into memory.
-        """
 
+            :param: floats : int | list | dict - A float or list of floats to  
+                load data from. Or a dictionary specifying floats and profiles
+                to read from the .nc file. 
+            :param: variables : str | list - An optional parameter to list variables
+                that the user would like included in the dataframe. If the variable is not
+                in the float passed then only the surface level of the profile will be included.
+
+            :return: float_data : pd - A dataframe with requested float data. 
+        """
         # Check that index files are in memory
+        if not self.download_settings.keep_index_in_memory: 
+            self.sprof_index = self.__load_sprof_dataframe()
+            self.prof_index = self.__load_prof_dataframe()
 
         # Check that passed float is inside of the dataframes
+        self.float_ids = floats
+        self.__validate_floats_kwarg()
 
-        # Download .nc files for passed floats
+        # Validate passed variables
+        self.float_variables = variables
+        if self.float_variables: self.__validate_float_variables_arg()
+
+        # Check if the user has passed only phys float variables
+        if self.float_variables is not None: 
+            phys_variables = ['TEMP', 'PSAL', 'PRES', 'CNDC']
+            only_phys = all(x in phys_variables for x in self.float_variables)
+        else: 
+            only_phys = False
 
-        # 
-        pass
+        # Download .nc files for passed floats
+        files = []
+        for wmoid in self.float_ids:  
+            # If the float is a phys float, or if the user has provided no variables 
+            # or only phys variables then then use the corresponding prof file
+            if (not self.float_stats.loc[self.float_stats['wmoid'] == wmoid, 'is_bgc'].values[0]) or (self.float_variables == None) or (only_phys): 
+                file_name = f'{wmoid}_prof.nc'
+                files.append(file_name)
+            # If the float is a bgc float it will have a corresponding sprof file
+            elif self.float_stats.loc[self.float_stats['wmoid'] == wmoid, 'is_bgc'].values[0]: 
```

Comment:
do we need this check? I don't think so; but if we did, we would need an else block

Reply:
you're right we shouldn't need this check, i'll change it to an else